services/comfy/Dockerfile

```diff
@@ -8,28 +8,24 @@ ENV DEBIAN_FRONTEND=noninteractive PIP_PREFER_BINARY=1
 
 RUN apt-get update && apt-get install -y git && apt-get clean
 
+RUN --mount=type=cache,target=/root/.cache/pip  \
+  --mount=type=bind,from=xformers,source=/wheel.whl,target=/xformers-0.0.21-cp310-cp310-linux_x86_64.whl \
+  pip install /xformers-0.0.21-cp310-cp310-linux_x86_64.whl
+
 ENV ROOT=/stable-diffusion
 RUN --mount=type=cache,target=/root/.cache/pip \
   git clone https://github.com/comfyanonymous/ComfyUI.git ${ROOT} && \
   cd ${ROOT} && \
   git checkout master && \
-  git reset --hard 884ea653c8d6fe19b3724f45a04a0d74cd881f2f && \
+  git reset --hard 91ed2815d542c96fdad75edba2205140de3cbba6 && \
   pip install -r requirements.txt
 
-
-RUN --mount=type=cache,target=/root/.cache/pip  \
-  --mount=type=bind,from=xformers,source=/wheel.whl,target=/xformers-0.0.21-cp310-cp310-linux_x86_64.whl \
-  pip install /xformers-0.0.21-cp310-cp310-linux_x86_64.whl
-
-
 WORKDIR ${ROOT}
 
-ARG BRANCH=master SHA=8607c2d42d10b0108de02528e813cc703e58813f
-RUN --mount=type=cache,target=/root/.cache/pip \
-  git fetch && \
-  git checkout ${BRANCH} && \
-  git reset --hard ${SHA} && \
-  pip install -r requirements.txt
+ADD https://github.com/madebyollin/taesd/raw/main/taesd_decoder.pth \
+  ${ROOT}/models/vae_approx/taesd_decoder.pth
+ADD https://github.com/madebyollin/taesd/raw/main/taesdxl_decoder.pth \
+  ${ROOT}/models/vae_approx/taesdxl_decoder.pth
```

Comment:
these be overridden by the mounts on container startup?

Reply:
I did test it and it worked. 

I believe the reason that this works, is because in the config https://github.com/AbdBarho/stable-diffusion-webui-docker/blob/master/services/comfy/extra_model_paths.yaml it only ever specifies subdirs of models. So, the models dir is never directly bindmounted (which would blow away these taesd decoders from models/vae_approx. 

If user *adds `vae_approx/` to the extra_model_paths.yaml*, then they could definitely add such a folder to mount and override these. Indeed this was my first iteration of my PR, but then I figured why not remove this manual step of fetching these and have the dockerfile auto install them.

Reply:
Isn't this something that should go into the download service? Seems more appropriate to me, and less likely to cause surprises.

Reply:
maybe? I am not really sure, most of the users want to use A1111, they maybe don't care about these files but they get downloaded regardless? not really sure...

Reply:
Fair enough. In that case, a download-comfy profile could be better?

Reply:
TAESD is useful for both comfy and a1111. I consider it to be an ideal compromise between overhead and quality, but afaik a1111 has a self installing option for it in its settings. My change is just to provide it for comfy. 

Expanding the download profile to have a comfy specific one is reasonable, I guess but kind of calls into question the whole approach of having a download profile tbh, I mean I get why it's a thing, but it's pretty awkward to require users to have to deal with choosing what they're suppose to be running. For example: if the workflow is always to run `docker compose up --profile download` prior to `docker compose up --profile auto`, to ensure the large files to download are present, then shouldn't we just take that download logic and make it idempotent and then insert that into the auto and comfy services themselves?

Reply:
@unphased maybe it would have been a lot easier to know that each of the files is only ~5MB each, I guess it would be totally fine to add them to the download service.

I was worried they are maybe a couple of gigabytes each, and that would be a problem.

Regarding
> make it idempotent and then insert that into the auto and comfy services themselves

That would be cool, the way we do it now is that we validate the checksums of the downloaded files, which takes a couple of minutes on a really beefy machine, and I would not want to run this on every statup.

The reason we validate the checksum is because the HF api drops the connection a lot and the downloads gets interrupted many times.
also because when we started this project, the files came from "unknown" resources, and made sense to validate them

maybe we can change this decision in the future, that would be another topic.